src/Components/RequireAuth/RequireAuth.jsx

```diff
@@ -0,0 +1,12 @@
+import { useLocation, Navigate, Outlet } from "react-router-dom";
+import { useAuth } from "../../Context";
+
+export const RequireAuth = () => {
+  const { authState: { userId } } = useAuth();
+  const location = useLocation();
+  return userId ? (
+    <Outlet />
+  ) : (
+    <Navigate to="/login" state={{ from: location }} replace />
```

Comment:
Yes this is to prevent the access to wishlist and cart page without logging in. The userId which I am using here comes as a response from the backend when I either sign up or login, so if I have a userID then I can access wishlist and cart page else if I click on wishlist or cart page without logging in I will not have the userID so the else part would be triggered which automatically directs to the login page from the current location(which is either the wishlist or the cart page)

Reply:
Okay understood. Nice implementation